manifests/client.pp

```diff
@@ -21,17 +23,27 @@
 # @author Scott Merrill <smerrill@covermymeds.com>
 # @note Copyright 2014 CoverMyMeds, unless otherwise noted
 #
+#
 class gluster::client (
-  Boolean $repo = $gluster::params::repo,
-  String $client_package = $gluster::params::client_package,
-  String $version = $gluster::params::version,
+  Boolean $repo             = $gluster::params::repo,
+  String $client_package    = $gluster::params::client_package,
+  String $version           = $gluster::params::version,
+  String $release           = $gluster::params::release,
+  Optional $repo_key_source = $gluster::params::repo_key_source,
 ) inherits gluster::params {
 
+  # if we manage the repository, we also need a GPG key
+  if $repo {
+    assert_type(Variant[Stdlib::Absolutepath, Stdlib::HTTPSUrl], $repo_key_source)
```

Comment:
I'm wondering about this pattern. It's hard to discover from reading the reference. Should the parameter be `Optional[Variant[Stdlib::Absolutepath, Stdlib::HTTPSUrl]]`? In that case maybe this could be simplified to `assert_type(NotUndef, $repo_key_source)`.

Reply:
I am not sure about the resource ordering, but I think this is redundant code. The value of ```$repo``` and  ```$repo_key_source``` is already defined at ```manifests/params.pp``` and we check the type at ```manifests/repo.pp```.

Reply:
I agree that it's redundant because `manifests/repo.pp` already checks it. That's a better approach than duplicating it here.